File: easy/2553_separate_the_digits_in_an_array/solution.py
from typing import List


class Solution:
    def separateDigits(self, nums: List[int]) -> List[int]:
        digits = []

        for num in nums:
            stack = []

            while num:
                stack.append(num % 10)
                num //= 10

            while stack:
                digits.append(stack.pop())

        return digits

File: easy/2553_separate_the_digits_in_an_array/Solution.java
class Solution {
  public int[] separateDigits(int[] nums) {
    int size = 0;
    for (final int num : nums) {
      size += Math.log10(num) + 1;
    }

    final int[] digits = new int[size];
    --size;

    for (int i = nums.length - 1; i > -1; --i) {
      int num = nums[i];

      while (num > 0) {
        digits[size] = num % 10;
        --size;
        num /= 10;
      }
    }

    return digits;
  }
}
